src/Pages/login.jsx

```diff
@@ -0,0 +1,111 @@
+import { event } from "jquery";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export default () => {
+  const navigate = useNavigate();
+  const [userName, setUserName] = useState(null);
+  const [password, setPassword] = useState(null);
+
+  const [isUsernameActivated, setUsernameActivated] = useState(false);
+  const [isPasswordActivated, setPasswordActivated] = useState(false);
+
+  const [unalert, setUnAlert] = useState(null);
+  const [pwalert, setPwAlert] = useState(null);
+
+  function handleUsername(event){
+    setUserName(event.target.value);
+    
+    checkUsername(userName);
+  
+  }
```

Comment:
We should look to take advantage of arrow functions in this case, and make use of react state hooks (useMemo, useCallback etc). State hooks will help in managing the state of a component more effectively with minimum conflicts and unnecessary rendering.

Reply:
Fixed e97eae75b859e2f1f70c09d6abe315d63c384f20